src/pydocstyle/parser.py

```diff
@@ -117,7 +119,35 @@ def is_public(self):
 
         This helps determine if it requires a docstring.
         """
-        return not self.name.startswith('_') or self.name.startswith('__')
+        module_name = Path(self.name).stem
+        return (
+            not self._is_inside_private_package() and
+            self._is_public_name(module_name)
+        )
+
+    def _is_inside_private_package(self):
+        """Return True if the module is inside a private package."""
+        path = Path(self.name).parent  # Ignore the actual module's name
+        syspath = [Path(p) for p in sys.path]  # Convert to pathlib.Path.
+
+        while path != path.parent and path not in syspath:  # Bail if we are at the root directory or in `PYTHONPATH`.
```

Comment:
Sorry for the error in my own code, but I guess this exceeds 80 chars? Just move the comment one line up.

Reply:
No worries, this should be resolved in https://github.com/PyCQA/pydocstyle/pull/494/commits/f8073657a6b0a63c89d33b2c2a7b2a9ffe25e306.